### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by taniacsilva**

### CODE REVIEW

File: life_expectancy/cleaning.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        life_exp (Pandas DataFrame): DataFrame with life expectancy cleaned data
+    """
+    script_dir = Path(__file__).resolve().parent
+    life_exp.to_csv(script_dir / "data"/ "pt_life_expectancy.csv", index=False)
+
+
+def main(region: str = 'PT') -> None:
+    """
+    Main function that load, cleans and save a cleaned version of the life expectancy data
+    """
```
Comment (luismelo1999) [inline life_expectancy/cleaning.py]: can you add the main() function args to the docstring

```suggestion
def main(region: str = 'PT') -> None:
    """
    Main function that load, cleans and save a cleaned version of the life expectancy data
    Args:
        region (str): String with region information
    """
```

File: life_expectancy/cleaning.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    Returns: 
+        life_exp (Pandas DataFrame): DataFrame with life expectancy cleaned data
+    """
+
+    # Split column with unit, sex, age, geo to multiple columns
+    columns = ["unit", "sex", "age", "region"]
+    life_exp_raw_data[columns] = (life_exp_raw_data.iloc[:,0].str.split(',', expand = True))
+    life_exp = life_exp_raw_data.drop(columns=life_exp_raw_data.columns[0])
+
+    # Filters only the data where region equal to PT
```
Comment (luismelo1999) [inline life_expectancy/cleaning.py]: in the comment, the filter is not mandatory to be 'PT' , I think

File: life_expectancy/cleaning.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        )
+
+    # Ensure year is an int
+    life_exp["year"] = life_exp["year"].str.replace(" ", "").astype(int)
+
+    # Ensure value is a float
+    life_exp["value"] = life_exp["value"].str.replace(r"[a-zA-Z\s]+", "",  regex=True)
+    life_exp["value"] = life_exp["value"].replace(":", np.NaN)
+    life_exp["value"] = life_exp["value"].astype(float)
+    life_exp = life_exp.dropna(subset=["value"])
```
Comment (luismelo1999) [inline life_expectancy/cleaning.py]: In these lines your comment is not completely in line with the code. You also do some data cleaning in the process

File: life_expectancy/cleaning.py (hunk 1 above)
Comment (taniacsilva) [reply life_expectancy/cleaning.py]: done, thanks for the correction 👍

File: life_expectancy/cleaning.py (hunk 3 above)
Comment (taniacsilva) [reply life_expectancy/cleaning.py]: done :)

**Review by luismelo1999 — APPROVED** [review]

**Pull request merged**